text/0715-construct-upgrade-support.md

```diff
@@ -0,0 +1,1251 @@
+|                │   CFN Change set    |                    |
+|                |       analysis      │                    |
+|                └─────────────────────┘                    |               
+|                           |                               |
+|                 (Optional)|──────────────┐                |
+|                           |              |                |
+|                           ▼              |                |
+|                ┌─────────────────────┐   |                |
+|                │ Custom Validations  |   |                |
+|                └─────────────────────┘   |                │
+|                           │              |                |
+|                           │──────────────┘                |
+|                           │                               |
+|           ┌───────────────┴───────────────┐               |
+|           ▼                               ▼               |
+|┌───────────────────────┐         ┌───────────────────────┐|   
+|│  Stack Refactoring    |         | Retain-Remove-Import  ||   
+||       Strategy        |         │        Strategy       ||   
+||           │           |         |            │          ||    
+||           ▼           |         |            ▼          ||    
+|| ┌───────────────────┐ |         |┌─────────────────────┐||                                 
+|| │ Execute Stack     | |         |│     Execute         |||  
+|| |    refactoring    | |         || Retain-Remove-Import||| 
+|| └───────────────────┘ |         |└─────────────────────┘|| 
+|└───────────────────────┘         └───────────────────────┘|
+|           │                               │               | 
+|           └───────────────────────────────┘               |
+|                          |                                |
+|                          ▼                                |
+|               ┌─────────────────────┐                     |
+|               │        Done         │                     |
+|               └─────────────────────┘                     |
+└───────────────────────────────────────────────────────────┘
+```
+
+## Settings
+
+You have a few settings available to control the behavior of the new CDK CLI command
+for construct upgrade feature.
+
+```sh
+cdk construct-upgrade --unstable=construct-upgrade [stacks] \
+  --target [id] \
+  [OPTIONS]
+```
+
+Arguments:
+
+- CDK stack ID (Optional)
+    The construct ID of the CDK stack from your app to synthesize.
+
+Options:
+
+- --target (Required):
+  The target construct and module you want to upgrade to. For example, `aws-cdk-lib.aws-dynamodb.TableV2` or `@aws-cdk/aws-ec2.VpcV2`
+  For fully supported construct upgrades, we can short hand it to `TableV2` or `VpcV2`.
+- --dry-run (Optional)
+  This flag is default to false. When the flag is set to true, the CLI acts as dry-run and will only trigger validation.
+- --require-approval (Optional)
```

Comment:
Nit, but this flag is poorly named. It should have been called `--require-confirmation`. We should deprecate the old flag, add the new one, and don't add any more copies of the silly name.